src/App/Router.php

```diff
@@ -283,7 +276,12 @@ private function getModuleClass()
 			$moduleClass = $routeInfo[1];
 			$this->parameters = $routeInfo[2];
 		} elseif ($routeInfo[0] === Dispatcher::METHOD_NOT_ALLOWED) {
-			throw new HTTPException\MethodNotAllowedException($this->l10n->t('Method not allowed for this module. Allowed method(s): %s', implode(', ', $routeInfo[1])));
+			if ($this->httpMethod === static::OPTIONS) {
```

Comment:
Wouldn't it be easier to just add the `OPTIONS` to the check in the line 144?

Reply:
No, because line 144 is the constructor and it should have as less logic as possible. And exceptions/errors aren't handled well there. Think about it like a real basic INIT method.

Reply:
I still don't get it. I feel like being in university and the explanation refers to something I should have learned in high school while I'm on an elementary school level.

Reply:
The `__constructor()` is called while creating/constructing the dynamic class. It should contain things to build the class, like variable instancing, parsing Environment variables, ... `__constructor()` are normally called very early in the execution flow where normally (second "normally" :D) not all dependencies and environments are finally set. So adding business logic into this special method could lead to unwanted side effects.

Therefore it's better to have `__constructor()`s as small&dumb as possible and add business logic at specific methods (`run()`, `generate()`, `dispatch()`, ...) where the caller can be sure that all specific environments/variables/catches are properly set.

Specific to our case:
The `Router()` constructor is called when we run `App-->runFrontend()` at the `index.php`. And we don't have any possibility to properly catch an exception here because it's called even before any Renderer/L10/... is available. So we must not add an Exception here.